models/optional_modules/core_web_vitals/bigquery/snowplow_web_vital_measurements.sql

```diff
@@ -0,0 +1,259 @@
+{{
+  config(
+    enabled=var("snowplow__enable_cwv", false) and target.type == 'bigquery' | as_bool(),
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with by_url_and_device as (
+
+  select distinct
+
+    page_url,
+    device_class,
+    'all' as country,
+    'last {{var('snowplow__cwv_days_to_measure')|string }} days' as time_period,
+    count(*) over (partition by page_url, device_class) as page_view_count,
+    nullif(percentile_cont(lcp, 0.{{ var('snowplow__cwv_percentile') }}) over (partition by page_url, device_class), 0) as lcp_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(fid, 0.{{ var('snowplow__cwv_percentile') }}) over (partition by page_url, device_class) as fid_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(cls, 0.{{ var('snowplow__cwv_percentile') }}) over (partition by page_url, device_class) as cls_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(ttfb, 0.{{ var('snowplow__cwv_percentile') }}) over (partition by page_url, device_class) as ttfb_{{ var('snowplow__cwv_percentile') }}p,
+    'by_url_and_device' as measurement_type
```

Comment:
In absolute awe that BigQuery doesn't support precentile_cont as a group by function, this explains why you did it the same way in Snowflake (although I still think the CUBE with a group by will make it much cleaner)